gossipd/routing.c

```diff
@@ -1529,10 +1556,11 @@ struct route_hop *get_route(const tal_t *ctx, struct routing_state *rstate,
 		hops[i].nodeid = n->id;
 		hops[i].amount = total_amount;
 		hops[i].delay = total_delay;
-		total_amount += connection_fee(c, total_amount);
+		total_amount += connection_fee(sourcenode, route[i], idx, total_amount);
```

Comment:
I'm wondering if we should simply skip this addition if `i==0` since that'd be the hop that we are in control over anyway. That'd avoid having to thread an additional argument in and do the check in another function.

Reply:
`connection_fee` is used in both this and `bfg_one_edge`.
We could argue for moving the check to `bfg_one_edge`, but properly, the source node charges no fee to itself for forwarding, so I thought, more appropriate at `connection_fee`.